.github/workflows/cicd.yaml

```diff
@@ -0,0 +1,56 @@
+---
+
+name: CI / CD Pipeline
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - '**/*.md'
+      - '**/*.rst'
+    branches:
+      - master
+      - '[0-9].*'
+  pull_request:
+    branches:
+      - master
+      - '[0-9].*'
+  schedule:
+    - cron: '0 1 * * *' # nightly build
+  workflow_dispatch:
+
+jobs:
+
+  build:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v4
+      - name: Set Java up n the runner
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+      - name: Install missing dependencies to container
+        run: |
+          sudo apt update
+          sudo apt install -y stunnel make git gcc maven
+      - name: Maven offline
```

Comment:
It would make sense to use the maven setup action:

```
    - name: Setup Maven Action
      uses: s4u/setup-maven-action@v1.8.0
      with:
        java-version: 8
```

to include Maven repo caching, avoiding repetitive artifact downloads.

Reply:
@mp911de What about the latest solution with caching included in the setup-java client? The [docs](https://github.com/actions/setup-java?tab=readme-ov-file#caching-packages-dependencies) mention that the caching is achieved in this way )by hashing the pom.xml) so I think this should suffice?